Utilities/StorageFactory/interface/StorageAccount.h

```diff
@@ -116,21 +153,61 @@ class StorageAccount {
     int m_value;
     
   };
-  
+
+  /**
+   * The `OpenLabelToken` provides callers of TFile::Open the ability to
+   * retrieve a separate set of statistics in the FrameworkJobReport (and
+   * other statistics services) for IO.  Usage looks like this:
+   *
+   * ```cpp
+   * TFile *file = nullptr;
+   * {
+   *   OpenLabelToken label = StorageAccount::setLabel("MixingModule");
+   *   file = TFile::Open("url://foo/bar");
+   * }
+   * ```
+   *
+   * Any IO activity associated with `file` above will get a separate
+   * aggregation in CMSSW statistics.  This allows, for example, bytes
+   * read from the mixing module to be differentiated from the primary source.
+   */
+  class OpenLabelToken {
+  friend class StorageAccount;
+
+  public:
+    OpenLabelToken(OpenLabelToken const&) = delete;
+    OpenLabelToken(OpenLabelToken &&) = default;
+    ~OpenLabelToken() {m_label = "";}
+
+  private:
+    OpenLabelToken(const std::string &label) {
+      m_label = label;
+    }
```

Comment:
This should take the previous value of the thread local and reset it in the destructor. That way if TBB tasks call other TBB tasks the code will be reset.